rs/execution_environment/src/execution_environment.rs

```diff
@@ -345,6 +347,7 @@ pub struct ExecutionEnvironment {
     // parallel and potentially reserving resources. It should be initialized to
     // the number of scheduler cores.
     resource_saturation_scaling: usize,
+    deallocator_thread: DeallocatorThread,
```

Comment:
As discussed offline, it would be better to flush this when dropping an `ExecutionEnvironment` during tests.

Reply:
Yes exactly. You can see the same pattern in the [state manager](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/state_manager/src/lib.rs?L852)

Reply:
The test failures on the PR also seem to suggest that flushing the thread on dropping `ExecutionEnvironment` for tests could be a good idea (note that I didn't look into the failures in detail; just saying this sounds like a plausible cause).

Reply:
Done.

I've tried my best to reproduce the CI test failure locally, to be able to test whether this addresses it (I restructured and added sleeps to `mark_checkpoint_files_readonly()`; increased the deallocator thread sleep duration), to no avail. I'll try restarting the CI job a few times if it doesn't fail, for good measure.

Reply:
I've restarted the CI pipeline 3 or 4 times and there was no failure. While that's hardly a proof that nothing got broken, it's the best I could do.